### PR TITLE
Feature/jjdd8fpo 7

### PR DESCRIPTION
**Pull request by lukaszwolek**

### CODE REVIEW

**Comment by isacademyci** [comment]
Build triggered for original commit.

**Comment by isacademyci** [comment]
Build started for original commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for original commit.

**Comment by isacademyci** [comment]
Build started for original commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

File: src/main/java/com/infoshareacademy/Book.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
     private String genre;
 
-    public static List<Book> book = new ArrayList<>();
+    public static ArrayList<Book> book = new ArrayList<>();
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/Book.java]: Dobre poprawiliście na złe... typujemy interfejsem.

File: src/main/java/com/infoshareacademy/BookList.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class BookList {
+    private static final Logger stdout = LoggerFactory.getLogger("CONSOLE_OUT");
+
+    ArrayList<Book> myList = new ArrayList<>();
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookList.java]: typujemy interfejsem

File: src/main/java/com/infoshareacademy/BookList.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
+package com.infoshareacademy;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class BookList {
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookList.java]: BookList? Do czego ta klasa ? jesli to część menu to nazwa jest kiepska, jesli implementacja listy  to po co skoro macie do wyboru do koloru list...

File: src/main/java/com/infoshareacademy/BookList.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+            if (record < books.size() + 1) stdout.info(record + ". " + book);
+            counter++;
+            record++;
+
+            if (counter > recordsLimit) {
+                stdout.info("\nType q to finish list, any to continue book list");
+                Scanner scanner = new Scanner(System.in);
+                String choice = scanner.next();
+                if (choice.equals("q")) {
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookList.java]: a jak choice bedzie nullem?

File: src/main/java/com/infoshareacademy/BookList.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        return;
+    }
+
+    private int chooseBookToPrint() {
+        stdout.info("\nWpisz numer ksiazki: ");
+        Scanner scanner = new Scanner(System.in);
+        int bookChoice = 0;
+        try {
+            bookChoice = scanner.nextInt() - 1;
+        } catch (NumberFormatException e) {
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookList.java]: https://docs.oracle.com/javase/7/docs/api/java/lang/NumberFormatException.html

Nie łapiemy RuntimeException!!! Za to będzie plank. :)

File: src/main/java/com/infoshareacademy/BookRepository.java [hunk 6]
```diff
@@ -1,13 +1,14 @@
 package com.infoshareacademy;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class BookRepository {
 
-    private static List<Book> bookRepository = Book.book;
+    private static ArrayList<Book> bookRepository = Book.book;
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookRepository.java]: jw

File: src/main/java/com/infoshareacademy/BookRepository.java [hunk 7]
```diff
@@ -1,13 +1,14 @@
...
 
 
-    public static List<Book> getBooks() {
+    public static ArrayList<Book> getBooks() {
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookRepository.java]: jw

File: src/main/java/com/infoshareacademy/Menu.java [hunk 8]
```diff
@@ -74,7 +79,8 @@ private void bookListMenu() {
 
         switch (getChoice(4)) {
             case 1: {
-                stdout.info(String.valueOf(BookRepository.getBooks()));
+                new BookList().printBooks(BookRepository.getBooks());
+                //stdout.info(String.valueOf(BookRepository.getBooks()));
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/Menu.java]: nie pushujemy kodu zakomentowanego, jak nie jest potrzebny to wywalamy

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

File: src/main/java/com/infoshareacademy/BookPrinter.java [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+                }
+            }
+        }
+        menuBookList();
+    }
+
+    private void menuBookList() {
+
+        stdout.info("\nWybierz: ");
+        stdout.info("\nc -       choose book");
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookPrinter.java]: pomieszane języki

File: src/main/java/com/infoshareacademy/BookPrinter.java [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+        stdout.info("\nm -       main menu");
+        stdout.info("\nq -       close application\n");
+
+        Scanner scanner = new Scanner(System.in);
+        String choice1 = scanner.next();
+
+        if (checkChoice1(choice1)) {
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookPrinter.java]: nazwy a znaczenie

File: src/main/java/com/infoshareacademy/BookPrinter.java [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+                }
+                case "c": {
+                    int temp = chooseBookToPrint();
+                    stdout.info(temp + 1 + ". " + String.valueOf(BookRepository.getBooks().get(temp)));
+                    menuBookList();
+                    break;
+                }
+            }
+        } else {
+            stdout.info("\nBledny wybor");
```
Comment (rafalmisiak) [inline src/main/java/com/infoshareacademy/BookPrinter.java]: pomieszane jezyki

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

File: src/main/java/com/infoshareacademy/BookPrinter.java (hunk 11 above)
Comment (rafalmisiak) [reply src/main/java/com/infoshareacademy/BookPrinter.java]: zostało main menu

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Pull request merged**